packages/compass-schema-validation/src/stores/store.spec.ts

```diff
@@ -42,7 +42,12 @@ const fakeDataService = {
       /* never resolves */
     }),
   isCancelError: () => false,
-  sample: () => [{ prop1: 'abc' }],
+  sampleCursor: () =>
+    ({
+      *[Symbol.asyncIterator]() {
+        yield* [{ prop1: 'abc' }];
+      },
```

Comment:
```suggestion
      async *[Symbol.asyncIterator]() {
        yield* [{ prop1: 'abc' }];
      },
```

Just so that this is actually the correct type, "next()" returns a promise instead of `{value, done}` directly.